deployments/constructs/carbyne-stack.ts

```diff
@@ -0,0 +1,577 @@
+      version: "3.6.6",
+      set: [
+        { name: "images.repository", value: "quay.io/minio/minio" },
+        { name: "images.tag", value: "RELEASE.2022-04-16T04-26-02Z" },
+        { name: "mcImage.repository", value: "quay.io/minio/mc" },
+        { name: "mcImage.tag", value: "RELEASE.2022-04-16T21-11-21Z" },
+        { name: "mode", value: "standalone" },
+        { name: "rootUser", value: "dummy-key" },
+        { name: "rootPassword", value: "dummy-secret" },
+        { name: "replicas", value: "1" },
+        { name: "persistence.enabled", value: "true" },
+        { name: "persistence.size", value: "1Gi" },
+        { name: "resources.limits.memory", value: "256Mi" },
+        { name: "resources.requests.memory", value: "256Mi" },
+      ],
+    });
+
+    dependables.push(minio);
+
+    const istioChart = new cdktf.TerraformAsset(this, "istio-chart-path", {
+      path: path.resolve(__dirname, "../charts/istio"),
+      type: cdktf.AssetType.DIRECTORY,
+    });
+
+    const istio = new helm.release.Release(this, `istio${config.idPostfix}`, {
+      dependsOn: [...dependables],
+      provider: config.helmProvider,
+      wait: true,
+      waitForJobs: true,
+      name: "cs-istio",
+      chart: `./${istioChart.path}`,
+      set: [
+        { name: "routes.hosts.amphora", value: "cs-amphora" },
+        { name: "routes.hosts.castor", value: "cs-castor" },
+      ],
+    });
+
+    dependables.push(istio);
+
+    if (config.isMaster) {
+      const etcd = new helm.release.Release(this, `etcd${config.idPostfix}`, {
+        dependsOn: [...dependables],
+        provider: config.helmProvider,
+        wait: true,
+        waitForJobs: true,
+        name: "cs-etcd",
+        chart: "etcd",
+        repository: "https://charts.bitnami.com/bitnami/",
+        version: "8.3.1",
+        set: [
+          { name: "auth.rbac.create", value: "false" },
+          { name: "service.type", value: "LoadBalancer" },
+        ],
+      });
+
+      dependables.push(etcd);
+    }
+
+    const redis = new helm.release.Release(this, `redis${config.idPostfix}`, {
+      name: "cs-redis",
+      repository: "https://charts.bitnami.com/bitnami/",
+      chart: "redis",
+      dependsOn: [...dependables],
+      provider: config.helmProvider,
+      wait: true,
+      waitForJobs: true,
+      set: [
+        { name: "architecture", value: "standalone" },
+        { name: "auth.enabled", value: "false" },
+        { name: "master.persistence.enabled", value: "true" },
+        { name: "master.persistence.size", value: "1Gi" },
+      ],
+    });
+
+    dependables.push(redis);
+
+    const castor = new helm.release.Release(this, `castor${config.idPostfix}`, {
+      dependsOn: [...dependables, minio, postgresDBMS],
+      provider: config.helmProvider,
+      wait: true,
+      waitForJobs: true,
+      name: "cs-castor",
+      chart: "castor",
+      repository: "oci://ghcr.io/carbynestack",
+      version: "0.2.0",
```

Comment:
with the manual deployment it is possible to easily override the versions (as well as the repository) used to have custom images deployed. I think this is an important feature especially when developing new functions.
Do we expect developers to use the manual deployment procedure or how can this be done using IaC?
Should there be a separate configuration file where CDKT is reading the used Carbyne Stack images from?

Reply:
issue created: https://github.com/carbynestack/carbynestack/issues/92